src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int nodeHash = key != null ? key.hashCode() : 0;
+        int nodeIndex = Math.abs(nodeHash) % table.length;
+        Node<K, V> node = table[nodeIndex];
+        if (node == null) {
+            table[nodeIndex] = new Node<>(nodeHash, key, value, null);
+            size++;
+            return;
+        }
+        while (true) {
+            if (node.key == key
+                    || node.key.equals(key)) {
```

Comment:
```suggestion
                    || node.key != null && node.key.equals(key)) {
```
add check for null, Because there would appear a NPE if node.key == null and key != null